### PR TITLE
Updates for poppy-simu integration

### PR DESCRIPTION
**Pull request by show0k**
- rename threejs to poppy-sim
- add an option "start_background_services" which run api servers as threads in background

### CODE REVIEW

**Comment by pierre-rouanet** [comment]
I think it's kind of weird to have as a simulator choice "poppy-sim". I think it should reflect the real "technological" choice (having only a visualisation in a web browser). Using V-REP could also fall into the poppy simulator category, no?

**Comment by show0k** [comment]
I made a typo, I wanted to write "poppy-simu".
Vrep is not a technology, it is a name of a software. It works uppon Qt, Bullet, written in c++ with an API in LUA and Python... 

For a common user the main technology stack used in a software a not important and mean nothing.

poppy-simu is the name of the 'simulator' we made for poppy robots which is able to work on a web browser. Threejs is not even a technology, it is a library made for using WebGL more easily.

Would you like to call pypot numpy because numpy array are used everywhere in pypot ? I really think it has no sense to call this viewer "threejs"...

Perhaps as you say it may be confusing for some people, but I think it is very well understood than V-REP is not made by the Poppy team. Another option would be "poppy-web-simu" ? We can make a poll.

**Comment by pierre-rouanet** [comment]
Ok let me try again because obviously I failed the first time ^^

Let's say at some point we develop another simulator with, for instance, physics but requiring more CPU. Will we call it poppy-other-simu?

**Comment by show0k** [comment]
Calling it poppy-viewer instead of poppy-simu could solve the problem (a future add of physics) ?
But if we add physics, it would be nice to have a checkbox to enable or disable the physics, and it will resolve the name finding.

**Comment by matthieu-lapeyre** [comment]
Or we could imagine a name without any technological aspect in it, like Poppy, pypot etc ... 
Something more poetic than poppy-js-3D-viewer-without-physics-yet-but-soon-poppy-simu

that's being said I have no idea for the moment :)

**Comment by pierre-rouanet** [comment]
What do you guys think about pj3vwpybsps? It actually stands for poppy-js-3D-viewer-without-physics-yet-but-soon-poppy-simu...

No? Sure?

We could keep the flower theme. Maybe with the idea of watching plant grow? A greenhouse? A botanic garden?

**Comment by show0k** [comment]
So, poppy-simu wasn't that bad, there is no technological aspects nor a long name...

**Comment by pierre-rouanet** [comment]
It still implies that it's the only simulation possibility for me. And the poetic aspect.. euh.. well... you see what I mean :-)

Shall we just make this a community discussion and just vote? we don't have good idea, let's people decide --> forum?

**Comment by show0k** [comment]
Ok, good idea. You make the topic ?

**Comment by show0k** [comment]
Let's go with the very funny poppy-simu (cf [poll in the forum](https://forum.poppy-project.org/t/cfc-looking-for-a-fun-name-for-the-new-poppy-simulator/2078)).
@pierre-rouanet @damiencaselli  ok to merge this ?

**Comment by ghost** [comment]
:+1:
